Adds guns

Kasuobes opened the pull request:
Supersedes #16500 

 - Added ion pistols, 8 shots no AOE, normal size, can be worn on belt and holstered.
 - Also added charge state sprites for ion rifle because fuck that proc.
 - Pistol sprites from d3athrow/vgstation13#14197

 - Added taser carbines, 10 shots, heavy taser and heavy shock beams, large size, can be worn on belt and back.
 - Added stun rifles, 12 stun shots, huge size, can be worn on back.
 - Rifle and carbine sprites recycled from existing ones


![](http://i.imgur.com/N1Yp3yW.png)

Kasuobes committed:
Adds ion pistols

Added ion pistols, 8 shots no AOE, normal size, can be worn on belt and holstered.

Kasuobes committed:
Adds taser carbine, stun rifle

Carbine: 10 heavy taser shots, large size
Rifle: 12 stun shots, huge size

BlueNexus commented:
If you're recycling pulse icons, use these instead
http://i.imgur.com/4wwNJTX.png

Has sprites for both your rifle and carbine.

https://www.dropbox.com/s/p1u1qgxta0x2jlt/things.dmi?dl=0

Also includes an ion carbine sprite, and the pulse sprites included can be used to replace our current pulse sprites

Kasuobes commented:
Send sprites.

BlueNexus commented:
Just did

Kasuobes commented:
Ugh.

3am, can't be damned to recolor more sprites. Looks good enough to not give my tired self eye cancer.

Maybe next time when I add more guns, or when I actually make pulse weaponry useful.

Kasuobes committed:
Adds pulse things

Kasuobes commented:
Added pulse things using @BlueNexus's sprites

![](http://i.imgur.com/QiqyEx5.png)

 - Reinvented pulse rifles. 36 pulse shots, high damage, can be fired in bursts. Huge size, can be worn on back.
 - Added pulse carbines, 24 pulse shots, medium damage, can be fired in bursts. Large size, can be worn on back and belt.
 - Added pulse pistols, 18 pulse shots, low damage, can be fired in bursts. Normal size, can be worn on belt and holstered.

Iamgoofball commented:
>when you need 165 sprites for 3 guns

Techhead0 commented:
Yeah, the energy weapon sprites are currently kinda abysmal that way.

Kasuobes commented:
Goofball made a comment that made sense.

I must be going mad.

comma commented:
Changelog isn't really needed since those aren't added anywhere in actual game :V

OhEmGee commented:
Why do pulse pistols only fire the 15-damage pulses? That's weaker than a laser carbine or even a small egun.

I really can't see them having any use at all. Imho they should fire at least mid pulses with very few charges.


...Unless you're working towards replacing the laser carbines in the emergency armory with pistols/carbines.

Kasuobes commented:
Weapon rebalancing coming soon(tm), if you're wondering why the egun has the same damage potential as laser carbines.

OhEmGee commented:
>rebalancing 
**_OH BOY_**


I'm sorry if I seem pensive, but rebalancing almost never ends well. Really, most of the time rebalancing PR's guarantee a lot of tug-o-war and arguing about the exact numbers. If one is made, expect nine more within the month.

Why try to fix something that isn't broken? At the moment, we are exactly as lethal as we want to be. Any nerfs to lethality would only serve to disrupt the game balance further.

Kasuobes commented:
![](http://cdn.crossmap.com/images/1/27/12736.jpg)

This vase has a hole in it. It looks pretty with that hole though. After a few years the hole became a standard feature for all vases.  Eventually someone comes by and patches that hole up with some new clay.

Question: at what point was the vase broken?

Kasuobes committed:
Merge branch 'dev' into HelpHelpImBeingOppressed

Techhead0 commented on code/modules/projectiles/guns/energy/pulse.dm:
```diff
@@ -3,17 +3,52 @@
+	max_shots = 18
+	w_class = ITEM_SIZE_NORMAL
+	requires_two_hands=1 //a bit heavy
+	burst_delay = 1
```
Looking at `burst_delay` (and `move_delay`), why do the smaller pulse guns shoot faster?

Kasuobes replied:
Smaller and lighter parts, faster reaction speed to diodes. Also fires a weaker beam so the internal heat compensation system limit the burst rate less.

comma commented:
Open PR when those are accessible in game.
We have enough useless admin-only weaponry.
